go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -369,7 +376,60 @@ func (dp diffPartition) getRowIter(ctx *sql.Context, ddb *doltdb.DoltDB, ss *sch
 	rd := diff.NewRowDiffer(ctx, fromSch, toSch, 1024)
 	rd.Start(ctx, fromData, toData)
 
-	src := diff.NewRowDiffSource(rd, joiner)
+	src := diff.NewRowDiffSource(rd, joiner, ctx)
+	src.AddInputRowConversion(fromConv, toConv)
+
+	return &diffRowItr{
+		ad:             rd,
+		diffSrc:        src,
+		joiner:         joiner,
+		sch:            joiner.GetSchema(),
+		fromCommitInfo: fromCmInfo,
+		toCommitInfo:   toCmInfo,
+	}, nil
+}
+
+// TODO: This is reused from commit_diff_table and we have likely changed it's behavior by not
+//       passing in super schema any more and changing this logic. Need to test and resolve this
+//       before releasing these changes.
+func (dp diffPartition) getRowIter(ctx *sql.Context, ddb *doltdb.DoltDB, joiner *rowconv.Joiner) (sql.RowIter, error) {
```

Comment:
dolt_diff and dolt_commit_diff should work the same with respect to their generated schemas. So you should just change it to match, either as part of this work or as a fast followup